#88 - Prevent reload of room already loaded

raghunandankavi2010 opened the pull request:
Prevent reload of room if room is already shown to the user upon selecting room in navigation drawer

raghunandankavi2010 committed:
Return id instead of messageid in getItemId MessageAdapter

raghunandankavi2010 committed:
Merge remote-tracking branch 'origin/develop' into local-stable

raghunandankavi2010 committed:
Kotlin way one liner

raghunandankavi2010 committed:
Merge remote-tracking branch 'upstream/develop' into develop

raghunandankavi2010 committed:
Merge remote-tracking branch 'upstream/develop' into local-stable

raghunandankavi2010 committed:
Merge remote-tracking branch 'upstream/develop' into develop

raghunandankavi2010 committed:
Merge remote-tracking branch 'upstream/develop' into local-stable

raghunandankavi2010 committed:
Merge remote-tracking branch 'upstream/develop' into develop

raghunandankavi2010 committed:
Merge remote-tracking branch 'upstream/develop' into local-stable

raghunandankavi2010 committed:
Merge branch 'local-stable' into room-selection-navdrawer

raghunandankavi2010 committed:
#88 - Prevent reload of room already loaded

Prevent reload of room if room is already shown to the user upon clicking room in nav drawer

raghunandankavi2010 committed:
Assign room number again if room is no selected

Assign room number again if room is no selected. Typo mistake fixed

Mauker1 commented on app/src/main/java/com/tristanwiley/chatse/chat/ChatActivity.kt:
```diff
@@ -181,6 +185,7 @@ class ChatActivity : AppCompatActivity(), ServiceConnection {
 
         //Load a default room
         loadChatFragment(ChatRoom(prefs.getString("lastRoomSite", Client.SITE_STACK_OVERFLOW), prefs.getInt("lastRoomNum", 15)))
+        room_num = 15
```
You can't assume that the default room for all users will be room 15. It may be true for us, but it won't be for the rest of the chatSE users.

Also, new users might not have joined into any rooms yet.

raghunandankavi2010 replied:
room_num = prefs.getInt("lastRoomNum", 15) would do for now?

Mauker1 commented on app/src/main/java/com/tristanwiley/chatse/chat/ChatActivity.kt:
```diff
@@ -450,6 +456,20 @@ class ChatActivity : AppCompatActivity(), ServiceConnection {
 
+    override fun onItemClick(chatRoom: ChatRoom) {
+
+        val roomNo = chatRoom.num
```
Try not to mix snake_case with camelCase on the same project. I'd rather only use camel case.

raghunandankavi2010 replied:
will fix the casing!.

Mauker1 commented on app/src/main/java/com/tristanwiley/chatse/chat/adapters/RoomAdapter.kt:
```diff
@@ -25,14 +25,30 @@ import org.jetbrains.anko.doAsync
+    private val roomClickListener:OnItemClickListener
+
+    init {
+        roomClickListener = context as OnItemClickListener
```
I'm not sure I like this. What if you pass a context that doesn't implement `OnItemClickListener`? For example: `getApplicationContext()`. @TimCastelijns I'd like your opinion on this.

raghunandankavi2010 replied:
what do you suggest we change here?

Mauker1 replied:
Perhaps we could explicitly require the listener interface on the constructor. But I'll wait Tim's opinion on this matter.

TimCastelijns replied:
Yes definitely avoid casting and pass the listener in the constructor.

Mauker1 commented on app/src/main/java/com/tristanwiley/chatse/chat/ChatActivity.kt:
```diff
@@ -65,6 +67,8 @@ class ChatActivity : AppCompatActivity(), ServiceConnection {
     private var isBound = false
     private val prefs = SharedPreferenceManager.sharedPreferences
 
+    private var room_num: Int? = -1
```
Why is this variable nullable?

raghunandankavi2010 replied:
this is a overlook on my part. will fix.

Mauker1 replied:
Ok!

Mauker1 reviewed (changes requested):
Take a look at the comments I made. We need to change those before merging.

raghunandankavi2010 committed:
Camel case fix/ null var fix/get roomnum from prefs

Fixes as suggested by mauker

Mauker1 commented on app/src/main/java/com/tristanwiley/chatse/chat/ChatActivity.kt:
```diff
@@ -67,7 +67,7 @@ class ChatActivity : AppCompatActivity(), ServiceConnection, RoomAdapter.OnItemC
     private val prefs = SharedPreferenceManager.sharedPreferences
 
-    private var room_num: Int? = -1
+    private var roomnum: Int = -1
```
roomNum :)

raghunandankavi2010 replied:
argh...

Mauker1 commented on app/src/main/java/com/tristanwiley/chatse/chat/ChatActivity.kt:
```diff
@@ -458,9 +458,9 @@ class ChatActivity : AppCompatActivity(), ServiceConnection, RoomAdapter.OnItemC
-             room_num = roomNo
+        val roomno = chatRoom.num
+        if ((roomnum!=-1) && roomnum != roomno) {
+            roomnum = roomno
```
Same thing as the other comment. roomNum and roomNo

raghunandankavi2010 committed:
Fix casing

Fix casing

TimCastelijns commented on app/src/main/java/com/tristanwiley/chatse/chat/ChatActivity.kt:
```diff
@@ -65,6 +67,8 @@ class ChatActivity : AppCompatActivity(), ServiceConnection {
     private var isBound = false
     private val prefs = SharedPreferenceManager.sharedPreferences
 
+    private var roomNum: Int = -1
```
Don't initialize with an invalid value. Make the variable nullable instead.

TimCastelijns commented on app/src/main/java/com/tristanwiley/chatse/chat/ChatActivity.kt:
```diff
@@ -181,6 +185,7 @@ class ChatActivity : AppCompatActivity(), ServiceConnection {
 
         //Load a default room
         loadChatFragment(ChatRoom(prefs.getString("lastRoomSite", Client.SITE_STACK_OVERFLOW), prefs.getInt("lastRoomNum", 15)))
+        roomNum= prefs.getInt("lastRoomNum", 15)
```
Don't hardcode the key. Use a const. Put it in SharedPreferenceManager with the others, maybe with the existing User keys, or if it is not related to the user, create a new class like RoomPreferenceKeys.

TimCastelijns commented on app/src/main/java/com/tristanwiley/chatse/chat/ChatActivity.kt:
```diff
@@ -450,6 +456,20 @@ class ChatActivity : AppCompatActivity(), ServiceConnection {
+    override fun onItemClick(chatRoom: ChatRoom) {
+
+        val roomNo = chatRoom.num
+        if ((roomNum!=-1) && roomNum != roomNo) {
```
This block is confusing because there are 2 variables that have almost the same name. Give them distinctive names like `currentRoomNr` and `clickedRoomNr` or something.

TimCastelijns commented on app/src/main/java/com/tristanwiley/chatse/chat/adapters/RoomAdapter.kt:
```diff
@@ -42,15 +58,16 @@ class RoomAdapter(val site: String, val list: MutableList<Room>, val context: Co
                 val roomNum = room.roomID.toInt()
-                (mContext as ChatActivity).loadChatFragment(ChatRoom(site, roomNum))
+                roomClickListener.onItemClick(ChatRoom(site, roomNum))
+               // (mContext as ChatActivity).loadChatFragment(ChatRoom(site, roomNum))
```
Don't commit commented code unless there is a reason for it. If there is a reason, it should be explained.

TimCastelijns commented on app/src/main/java/com/tristanwiley/chatse/chat/adapters/RoomAdapter.kt:
```diff
@@ -25,14 +25,30 @@ import org.jetbrains.anko.doAsync
     override fun getItemCount() = list.size
 
 
+
```
Bunch of blank lines

raghunandankavi2010 committed:
Moving keys to constants and avoiding hard code keys

Moving keys to constants and avoiding hard coded keys by creating a RoomPreferenceKeys file

raghunandankavi2010 committed:
Listener param in constructor of RoomAdapter

Listener param in  constructor of RoomAdapter to avoid casting to context

raghunandankavi2010 committed:
Few suggestions by Tim incorporated

Remove unused imports . Use constants and avoiding hardcode keys. Pass listener param to RoomAdapter. Renaming of room numbers and making currentRoomNum nullable.

raghunandankavi2010 committed:
Remove unnecessary @ChatActivity 

Having this will do. So remove @ChatActivity

raghunandankavi2010 commented:
@Mauker1 fixed. can you take a look again!.

TimCastelijns commented on app/src/main/java/com/tristanwiley/chatse/chat/ChatActivity.kt:
```diff
@@ -458,13 +459,13 @@ class ChatActivity : AppCompatActivity(), ServiceConnection, RoomAdapter.OnItemC
                 addChatFragment(createChatFragment(chatRoom))
             }
-            prefs.edit().putString("lastRoomSite", chatRoom.site).putInt("lastRoomNum",roomNo).apply()
+            prefs.edit().putString("lastRoomSite", chatRoom.site).putInt("lastRoomNum",clickedRoomNo).apply()
```
Use the constants here as well. Currently here we have lastRoomSite and the constant is last_room_site

raghunandankavi2010 replied:
Oops. Missed that. Will fix

raghunandankavi2010 committed:
Use of constants instead of hard code keys. Assign Room number on click

Remove use of hard coded keys. Assign clicked room number to current room number if room name clicked is not same as current room number.

raghunandankavi2010 committed:
Yet another place replacing hard coded keys with constants.

Yet another place replacing hard coded keys with constants.
